### PR TITLE
reimplementation

### PR DESCRIPTION
**Pull request by decanus**

### CODE REVIEW

File: Sources/BeaconChain/DataStructures/State/Eth1Data.swift [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import Foundation
 
-struct Eth1Data {
+struct Eth1Data: Equatable {
     let depositRoot: Data
     let blockHash: Data
+
+    public static func == (lhs: Eth1Data, rhs: Eth1Data) -> Bool {
```
Comment (alejandro-isaza) [inline Sources/BeaconChain/DataStructures/State/Eth1Data.swift]: No need to add this, Swift 4 will synthesize.

File: Sources/BeaconChain/Constants.swift [hunk 2]
```diff
@@ -1,46 +1,54 @@
 import Foundation
 
+// Misc
 let SHARD_COUNT = UInt64(2**10)
 let TARGET_COMMITTEE_SIZE = UInt64(2**7)
-let EJECTION_BALANCE = UInt64((2**4) * Int(1e9))
+let EJECTION_BALANCE = UInt64(2**4 * UInt64(1e9))
```
Comment (alejandro-isaza) [inline Sources/BeaconChain/Constants.swift]: In Swift the naming convention for constants is the same as variables. Maybe just encapsulate in `struct Constants{...}`. They should not be all uppercase.

File: Sources/BeaconChain/Enums/StatusFlag.swift [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import Foundation
+
+enum StatusFlag: UInt64 {
+    case INITIATED_EXIT = 1
```
Comment (alejandro-isaza) [inline Sources/BeaconChain/Enums/StatusFlag.swift]:
```suggestion
    case initiatedExit = 1
```

File: Sources/BeaconChain/Enums/StatusFlag.swift [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    case WITHDRAWABLE = 2
```
Comment (alejandro-isaza) [inline Sources/BeaconChain/Enums/StatusFlag.swift]:
```suggestion
    case withdrawable = 2
```

File: Sources/BeaconChain/Constants.swift (hunk 2 above)
Comment (decanus) [reply Sources/BeaconChain/Constants.swift]: This file will be changed to a configuration file later on.

File: Sources/BeaconChain/DataStructures/State/Eth1Data.swift (hunk 1 above)
Comment (decanus) [reply Sources/BeaconChain/DataStructures/State/Eth1Data.swift]: I'm using swift 4.2 and it definitely seems to require it.

**Pull request merged**